lib/src/Pos/Logic/Full.hs

```diff
@@ -162,7 +163,15 @@ logicFull genesisConfig txpConfig ourStakeholderId securityParams jsonLogTx =
             { toKey = pure . Tagged . hash . taTx . getTxMsgContents
             , handleInv = \(Tagged txId) -> not . HM.member txId . _mpLocalTxs <$> withTxpLocalData getMemPool
             , handleReq = \(Tagged txId) -> fmap TxMsgContents . HM.lookup txId . _mpLocalTxs <$> withTxpLocalData getMemPool
-            , handleData = \(TxMsgContents txAux) -> Txp.handleTxDo genesisConfig txpConfig jsonLogTx txAux
+            , handleData = \(TxMsgContents txAux) -> do
+                currentEos <- getEpochOrSlot <$> getTipHeader
+                let txValRules = configTxValRules $ genesisConfig
+                let currentTxValRules = (TxValidationRules
+                                            (getAddressAttribCutoff txValRules)
+                                            currentEos
+                                            (getAddressAttribSize txValRules)
+                                            (getTxAttribSize txValRules))
+                Txp.handleTxDo genesisConfig currentTxValRules txpConfig jsonLogTx txAux
```

Comment:
❓ I'm confused by this part. We seem to pull the `TxValidationRules` out of the `genesisConfig`, update with `currentEos`, and pass it as a separate argument to a function which also takes `genesisConfig`. Is it possibly bad to have two datatypes of the same type (`TxValidationRules`) which have almost the same data, but are slightly different? What if we just updated the `genesisConfig` and passed that?

Reply:
Good catch! I'm currently looking into this, I'll update here when I'm finished.

Reply:
The `TxValidationRules` turned out to be unnecessary and I successfully removed it.